app/Traits/CategoryFilter.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Traits;
+
+use Illuminate\Support\Facades\Auth;
+use App\Models\Category;
+use Illuminate\Support\Facades\DB;
+use App\Models\Quiz;
+use App\Models\User;
+
+trait CategoryFilter
+{
+    protected function filter($query)
+    {
+        $taken_quizzes_id = Quiz::join('quizzes_taken', 'quizzes.id', '=', 'quizzes_taken.quiz_id')
+                            ->where('quizzes_taken.user_id', Auth::user()->id)
+                            ->get()
+                            ->pluck('category_id')
+                            ->all();
+
+        if ($query['filter'] === "Taken"){
+            $categories = Category::wherein('id', $taken_quizzes_id);
+        } elseif ($query['filter'] === "Not Taken"){
+            $categories = Category::whereNotIn('id', $taken_quizzes_id);
+        }
```

Comment:
please use `else` here just in case there is no data for filter. The category will fetch a 'default' data